app/domain/authentication/constraints/permitted_constraint.rb

```diff
@@ -0,0 +1,24 @@
+module Authentication
+  module Constraints
+
+    # This constraint validates that only the initialized restrictions are permitted.
+    # So, no other restrictions can be found in the given restrictions when calling validate(given_restrictions).
+    class PermittedConstraint
+
+      def initialize(permitted:)
+        @permitted = permitted
+      end
+
+      def validate(resource_restrictions:)
+        not_supported_restrictions = resource_restrictions - @permitted
+        not_supported_restrictions.each do |not_supported_restriction_name|
```

Comment:
Maybe instead of go over the list, we should just validate it is not empty and to throw 1 exception which contains all the not supported restrictions ?
For example in your case if we have 2 not supported restrictions we will throw an exception on the first one,
In my suggestion we will throw an exception with all the un supported restrictions

Reply:
done